profile/src/main/java/rdx/works/profile/domain/GenerateProfileUseCase.kt

```diff
@@ -51,15 +50,16 @@ class GenerateProfileUseCase @Inject constructor(
         mnemonicWithPassphrase: MnemonicWithPassphrase,
         accounts: Accounts
     ): Profile {
-        val networkId = accounts().firstOrNull()?.networkId ?: NetworkId.MAINNET
+        val accountsList = accounts.asList()
+        val networkId = accountsList.firstOrNull()?.networkId ?: NetworkId.MAINNET
         return when (val state = profileRepository.profileState.first()) {
             is ProfileState.Restored -> state.profile
             else -> withContext(defaultDispatcher) {
                 val profile = Profile.init(
                     deviceFactorSource = deviceFactorSource,
                     creatingDeviceName = deviceInfoRepository.getDeviceInfo().displayName
                 ).addAccounts(
-                    accounts = accounts(),
+                    accounts = accountsList,
```

Comment:
perhaps saver to just convert to list here, and not on the lines above. ensuring that no one accidentally modified `accountsList`.

Reply:
Yes but that means that the identified struct would be turned into a list twice. The reason I created `accountsList` is because I needed to export as list before. That is one of the many downgrades to code quality I had to do in order to change to this new api.